source/common/decompressor/zlib_decompressor_impl.cc

```diff
@@ -45,19 +45,12 @@ void ZlibDecompressorImpl::decompress(const Buffer::Instance& input_buffer,
     zstream_ptr_->next_in = static_cast<Bytef*>(input_slice.mem_);
     while (inflateNext()) {
       if (zstream_ptr_->avail_out == 0) {
-        output_buffer.add(static_cast<void*>(chunk_char_ptr_.get()),
-                          chunk_size_ - zstream_ptr_->avail_out);
-        chunk_char_ptr_ = std::make_unique<unsigned char[]>(chunk_size_);
-        zstream_ptr_->avail_out = chunk_size_;
-        zstream_ptr_->next_out = chunk_char_ptr_.get();
+        updateOutput(output_buffer);
       }
     }
   }
 
-  const uint64_t n_output{chunk_size_ - zstream_ptr_->avail_out};
-  if (n_output > 0) {
-    output_buffer.add(static_cast<void*>(chunk_char_ptr_.get()), n_output);
-  }
+  updateOutput(output_buffer);
```

Comment:
Interesting, can we have a short explanation of why the previous code can't solve to unbreak decompressor internal state.